src/api/rewardsHistory.ts

```diff
@@ -15,36 +15,50 @@ async function extractVaultUserRewards(
         user: allocatorAddress.toLowerCase(),
         dateFrom: Math.floor(dateFrom.getTime() / 1000).toString(),
     };
-    const responseRewards = await connector.graphqlRequest({
-        type: 'api',
-        op: 'UserRewards',
-        query: `query UserRewards($user: String!, $vaultAddress: String!, $dateFrom: DateAsTimestamp!) { userRewards(user: $user, vaultAddress: $vaultAddress, dateFrom: $dateFrom) { date, sumRewards, }}`,
-        variables: vars_getRewards,
-        onSuccess: function (value: Response): Response {
-            return value;
-        },
-        // eslint-disable-next-line @typescript-eslint/no-explicit-any
-        onError: function (reason: any): PromiseLike<never> {
-            throw new Error(`Failed to get rewards from Stakewise: ${reason}`);
-        },
-    });
+    try {
+        const responseRewards = await connector.graphqlRequest({
+            type: 'api',
+            op: 'UserRewards',
+            query: `query UserRewards($user: String!, $vaultAddress: String!, $dateFrom: DateAsTimestamp!) { userRewards(user: $user, vaultAddress: $vaultAddress, dateFrom: $dateFrom) { date, sumRewards, }}`,
+            variables: vars_getRewards,
+            onSuccess: function (value: Response): Response {
+                return value;
+            },
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+            onError: function (reason: any): PromiseLike<never> {
+                throw new Error(`Failed to get rewards from Stakewise: ${reason}`);
+            },
+        });
+        if (!responseRewards.ok) {
```

Comment:
This check (for status code and .errors field length) appears  to be consistent across all response handlers for all methods, maybe we could extract it to a method of connector:

```typescript
 validateResponse = (response: Response): Promise<data: object> => {
    if (!response.ok) {
       throw new Error(`Invalid response from Stakewise URI: ${response.url}`);
    }

    const responseData = await response.json();

        if (responseData.errors && responseData.errors.length) {
            throw new Error(responseData.errors[0].message);
        }

    return responseData
}


So then no need to repeat check-n-throw logic in all methods. Perhaps, this error handling logic can be even moved to run as callback of `fetch()` call inside `graphqlRequest` method of a connector. In that case, all future requests will be also validated automatically.